src/components/SearchBar.js

```diff
@@ -0,0 +1,62 @@
+import { getSearchData } from "utils/data";
+import { dispatch } from "utils/dispatch";
+
+export class SearchBar {
+  searchBar = document.getElementById("search-bar");
+  autoCompleteElement = document.getElementById("search-bar-autocomplete");
+
+  constructor() {
+    this.searchBar.addEventListener("input", () => {
+      dispatch.call("search-bar-autocomplete", this, this.searchBar.value);
+    });
+    this.searchBar.addEventListener("blur", () => {
+      dispatch.call("remove-autocompete-dropdown", this);
+    });
+    this.searchBar.addEventListener("focus", () => {
+      this.searchBar.value = "";
+    });
+    dispatch.on("search-bar-autocomplete", this.autoComplete);
+    dispatch.on("remove-autocompete-dropdown", this.removeAutocomplete);
+    dispatch.on("search-fetch-data-reject", (err) => console.error(err));
+  }
+
+  removeAutocomplete() {
+    setTimeout(() => {
+      this.autoCompleteElement.innerHTML = "";
+      this.searchBar.value =
+        "\u{01f50d}  Enter Nation, state, province, city or zipcode";
+    }, 400);
+  }
+
+  async autoComplete(str) {
+    if (str.length > 3) {
+      const res = await getSearchData(str);
```

Comment:
Maybe wrap the `await` in a `try / catch` block to handle any errors from the geocoder?

Reply:
Did this